packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts

```diff
@@ -197,8 +202,18 @@ function PatchedMutationObserver(
     return observer;
 }
 
-function patchedDisconnect(this: MutationObserver): void {
+function patchedDisconnect(this: PatchedMutationObserver): void {
     originalDisconnect.call(this);
+    // Remove all instances of this MutationObserver from the target Nodes
+    if (!isUndefined(this.observerLookups)) {
+        this.observerLookups.forEach(observerLookup => {
```

Comment:
use the ForEach function that you can import from `shared/language`

Reply:
we use a very similar trick for the backpointers in our reactive mechanism, so, I'm fine with this algo, probably changing a little bit the names... so it is not confusing, but I can help with that.

Reply:
in the reactivity, we use the concept of a collection of records (which in this case is what the observer keeps track of, while a record keeps track of the actual observers associated to a node, and it is own by the node. I feel that the confusing part of the code is about the ownership of these pieces.